common/interpreter/invocation.json

```diff
@@ -0,0 +1,114 @@
+    "id": "http://tradenet.org/pdo/contract/invocation#",
+    "definitions": {
+        "InvocationRequest": {
+            "id": "#invocation-request",
+            "type": "object",
+            "description" : [
+                "invocation request passed into the contract interpreter"
+            ],
+            "properties": {
+                "Method": {
+                    "description": [ "name of the method to invoke" ],
+                    "type": "string",
+                    "required": true
+                },
+                "PositionalParameters" : {
+                    "description": [ "array of parameters passed by position" ],
+                    "type": "array",
+                    "items": {
+                        "type": ["number","string","boolean","object","array"],
+                        "minItems": 0,
+                        "uniqueItems": false,
+                        "default": []
+                    },
+                    "required": true
+                },
+                "KeywordParameters" : {
+                    "description": [ "key/value parameters passed to the method" ],
+                    "type": "object",
+                    "required": true
+                }
+            }
+        },
+        "InvocationResponse" : {
+            "id" : "#invocation-response",
+            "description" : [
+                "encoded response returned from processing the invocation request"
+            ],
+            "type" : "object",
+            "properties" : {
+                "Status" : {
+                    "description" : [ "success or failure of the method invocation" ],
+                    "type" : "boolean",
+                    "required" : true
+                },
+                "Response" : {
+                    "description" : [ "value returned from the invocation" ],
+                    "type": ["number","string","boolean","object","array"],
+                    "required" : true
+                },
+                "StateChanged" : {
+                    "description" : [ "flag to indicate that the state was modified" ],
+                    "type" : "boolean",
+                    "required" : true
+                },
+                "Dependencies": {
+                    "description": [ "List of dependent contract commits" ],
+                    "type": "array",
+                    "items": {
+                        "$ref": "#/pdo/basetypes/state-reference",
+                        "minItems": 0,
+                        "uniqueItems": true,
+                        "default": []
+                    },
+                    "required" : true
+                }
+            }
+        },
+        "InvocationEnvironment" : {
+            "id" : "#environment",
+            "description" : [ "encoded enviroment information" ],
+            "type" : "object",
+            "properties" : {
+                "ContractID" : {
```

Comment:
from a readability/understandability perspective, would be nice to group fields into contract-static stuff (e.,g cc-name/code, creator, ..), invocation input specific (originator/message) and output (state)?  Also, why isn't dependencies also part of this?  (at least, to me it seems InvocationEnvironment is more or less exactly the commitment of a result of a transaction, with the dependencies the only missing piece?)

BTW: Both probably a bit late now given that it would require a fair bit work to change the code, but i guess i had to give some 5cents  ;-)

Reply:
dependencies are, by definition, a part of the result. that is, dependencies communicate to the ledger that the current transaction cannot be committed until another has also been committed. the contract invocation CREATES those dependencies.